### PR TITLE
use SEPAL viz params embeded in the Image metadata to display asset

### LINKED ISSUE #278: customize addLayer to aligne it on the SEPAL viz parameters
https://sepal-ui.readthedocs.io/en/latest/tutorials/create_asset.html for GEE asset

### PR DESCRIPTION
**Pull request by 12rambau**
Fix #278  

Here is an example file to discover the display options: users/bornToBeAlive/imageViZExample

It is embeding all the possible combination of viz params that SEPAL produces. 

to get them all displayed: 
```python
from sepal_ui import mapping as sm
import ee

image = ee.Image("users/bornToBeAlive/imageViZExample")
sm.SepalMap.get_viz_params(image)
```

display this asset in a map using one of the available viz params: 

```python
from sepal_ui import mapping as sm 
import ee

m = sm.SepalMap()
image = ee.Image("users/bornToBeAlive/imageViZExample")
m.addLayer(image, {}, "test", viz_name="NDWI")
m.zoom_ee_object(image.geometry())
m
```
Capture d’écran 2021-12-03 à 22 55 34

I still cannot manage to deal with inverted bands (in `NDWI harmonics`), if you have any idea be my guest ;-) 

## EDIT

managed inverted and hsv images: 

```python
from sepal_ui import mapping as sm 
import ee

ee.Initialize() 

m = sm.SepalMap()
image = ee.Image("users/bornToBeAlive/sepal_ui_test/imageViZExample")
m.addLayer(image, {}, "test", viz_name="NDWI harmonics")
m.zoom_ee_object(image.geometry())
m
```

Capture d’écran 2021-12-06 à 17 22 49

### CODE REVIEW

**Review by dfguerrerom — APPROVED** [review]

File: sepal_ui/mapping/mapping.py [hunk 1]
```diff
@@ -569,6 +570,71 @@ def add_colorbar(
 
         return
 
+    def addLayer(self, *args, viz_name=False, **kwargs):
```
[the remark this answers is not in the document]
Comment (12rambau) [reply sepal_ui/mapping/mapping.py:573]: I'm actually forced to use *args in combination with *kwargs as `addLayer` have positional arguments :
```python
m.addLayer(asset, {}, 'toto') 
```

The only solution would be to retrieve all the existing poisitional agrument from addLayer with the risk of being out of sync if something is changed on geemap side

[the remark this answers is not in the document]
Comment (dfguerrerom) [reply sepal_ui/mapping/mapping.py:573]: geemap `addLayer` only has one positional parameter: the ee_object. 
Well, that's not harmful, I've approved my review so you can go with the merge :)

[the remark this answers is not in the document]
Comment (12rambau) [reply sepal_ui/mapping/mapping.py:573]: I'll have a look tomorrow morning, if I can simplify the readability, I'm in

**Comment by 12rambau** [comment]
@dfguerrerom better ?

**Pull request merged**